cpp/src/groupby/sort/group_reductions.hpp

```diff
@@ -42,9 +54,40 @@ std::unique_ptr<column> group_sum(column_view const& values,
                                   rmm::cuda_stream_view stream,
                                   rmm::mr::device_memory_resource* mr);
 
+/**
+ * @brief Internal API to calculate groupwise product
+ *
+ * @code{.pseudo}
+ * values        = [2, 1, 4, -1, -2, <NA>, 4, <NA>]
+ * group_labels  = [0, 0, 0,  1,  1,    2, 2,    3]
+ * num_groups    = 4
+ *
+ * group_product = [6, 2, 4, <NA>]
+ * @endcode
+ *
+ * @param values Grouped values to get product of
+ * @param num_groups Number of groups
+ * @param group_labels ID of group that the corresponding value belongs to
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ */
+std::unique_ptr<column> group_product(column_view const& values,
+                                      size_type num_groups,
+                                      cudf::device_span<size_type const> group_labels,
+                                      rmm::cuda_stream_view stream,
+                                      rmm::mr::device_memory_resource* mr);
```

Comment:
Should these parameters have defaults (`rmm::cuda_stream_default` and `rmm::mr::get_current_device_resource()`) or are you assuming this function will only ever be called internally from functions that have already have a stream and memory resource?

Reply:
Assuming this will be called internally only.